lib/avromatic/model/types/map_type.rb

```diff
@@ -68,6 +68,10 @@ def serialize(value, strict:)
             end
           end
         end
+
+        def referenced_models
+          [key_type, value_type].tap(&:uniq!).flat_map(&:referenced_models).tap(&:uniq!)
```

Comment:
nit: Doesn't seem like the extra `tap(&:uniq!)` buys us much
```suggestion
          [key_type, value_type].flat_map(&:referenced_models).tap(&:uniq!)
```

Reply:
It's my lazy way of performing the search twice for the same type. I'll leave it as-is if you don't have strong feelings about it.

Reply:
I don't have strong feelings about it. Avro [maps](https://avro.apache.org/docs/current/spec.html#Maps) always have string keys so `key_type.referenced_models` should always return the empty array anyway.

Reply:
Oh interesting! As a non-native speaker, it's my understanding that the spec doesn't force keys to be strings, but google says otherwise. https://github.com/apache/avro/pull/17/files even shows "non-string" maps being encoded as arrays to work around this.

I'm going to include this info in the PR.